### PR TITLE
Support dawn of heroes

### PR DESCRIPTION
**Pull request by akshualy**
# Summary

After a bunch of requests for support for it, I decided to implement support for the new temporary game mode with it's own battle pass.

# Changes
- Added option to configure which mode the bot will play (currently only "normal" and "dawn of heroes")
- Added traits for dawn of heroes in a separate folder and separate class
- Support the second post game screen dawn of heroes implements

We'll have to remove the mode specific things in 3-4 weeks, but we can leave in the option to select different modes. Don't have a preference.

## Unrelated general changes and improvements
- Increase rate at which the bot buys XP
- Fix the carousel check since they swapped the images of being on board and being in carousel
- Increase readability on errors by waiting 10 seconds before exiting Alune if an error has been raised
- Log which surrender mode is currently selected and for which type of pass it is recommended

### CODE REVIEW

**Comment by Jokiblack** [comment]
Ooo this is cool, what xp increase margin did you put?

**Comment by akshualy** [comment]
> Ooo this is cool, what xp increase margin did you put?

https://github.com/TeamFightTacticsBots/Alune/pull/61/commits/d24e5fba6c5a2144fbf25031cc9b7fa61b2710db

File: alune/screen.py [hunk 1]
```diff
@@ -48,6 +49,52 @@ def get_button_on_screen(
...
+    Args:
+        path: The relative or absolute path to the image to be found.
+
+    Returns:
+        The image or none if it does not exist.
+    """
+    image_to_find = cv2.imread(path, 0)
+
+    if image_to_find is None:
+        logger.warning(f"The image {path} does not exist on the system " f"or we do not have permission to read it.")
```
Comment (Kyrluckechuck) [inline alune/screen.py:65]: Why are we splitting the string up (via `" f"`)? 🤔

**Review by Kyrluckechuck — APPROVED** [review]
Looks fantastic! A minor nit about some semantics but non-blocking as it still works, obviously

File: alune/screen.py (hunk 1 above)
Comment (akshualy) [reply alune/screen.py:65]: Not intentional, my brain has a little different opinion than black auto format sometimes

**Pull request merged**